scripts/releaseChangelog.js

```diff
@@ -0,0 +1,243 @@
+      const searchPullRequestId = commitsItem.commit.message.match(/\(#([0-9]+)\)/);
+      if (!searchPullRequestId || !searchPullRequestId[1]) {
+        return;
+      }
+
+      const {
+        data: { body: bodyMessage },
+      } = await octokit.request('GET /repos/{owner}/{repo}/pulls/{pull_number}', {
+        owner: GIT_ORGANIZATION,
+        repo: GIT_REPO,
+        pull_number: Number(searchPullRequestId[1]),
+      });
+
+      if (!bodyMessage) {
+        return;
+      }
+
+      const changelogMotif = '# changelog';
+      const changelogIndex = bodyMessage.toLowerCase().indexOf(changelogMotif);
+      if (changelogIndex >= 0) {
+        changeLogMessages.push(
+          `From https://github.com/${GIT_ORGANIZATION}/${GIT_REPO}/pull/${
+            searchPullRequestId[1]
+          }\n${bodyMessage.slice(changelogIndex + changelogMotif.length)}`,
+        );
+      }
+    }),
+  );
+
+  // Get all the authors of the release
+  const authors = Array.from(
+    new Set(
+      commitsItems.map((commitsItem) => {
+        return commitsItem.author.login;
+      }),
+    ),
+  );
+
+  // Dispatch commits in different sections
+  const changeCommits = [];
+  const coreCommits = [];
+  const docsCommits = [];
+  const otherCommits = [];
+
+  commitsItems.forEach((commitItem) => {
+    const tag = parseTags(commitItem.commit.message);
+    switch (tag) {
+      case 'DataGrid':
+      case 'DataGridPro':
+      case 'l10n':
+      case '118n':
+        changeCommits.push(commitItem);
+        break;
+      case 'docs':
+        docsCommits.push(commitItem);
+        break;
+      case 'core':
+        coreCommits.push(commitItem);
+        break;
+      default:
+        otherCommits.push(commitItem);
+        break;
+    }
+  });
+
+  // Helper to print a list of commits in a section of the changelog
+  const logChangelogSection = (commitsList, header) => {
+    if (commitsList.length === 0) {
+      return '';
+    }
+
+    const sortedCommits = commitsList.sort((a, b) => {
+      const aTags = parseTags(a.commit.message);
+      const bTags = parseTags(b.commit.message);
+      if (aTags === bTags) {
+        return a.commit.message < b.commit.message ? -1 : 1;
+      }
+      return aTags.localeCompare(bTags);
+    });
+    return `${header}
+
+${sortedCommits
+  .sort()
+  .map((commitItem) => `- ${commitItem.commit.message.split('\n')[0]} @${commitItem.author.login}`)
+  .join('\n')}
+`;
+  };
+
+  const nowFormatted = new Date().toLocaleDateString('en-US', {
+    month: 'short',
+    day: 'numeric',
+    year: 'numeric',
+  });
+
+  const changelog = `
+## TODO RELEASE NAME
+<!-- generated comparing ${lastRelease}..${release} -->
+_${nowFormatted}_
+
+A big thanks to the ${
+    authors.length
+  } contributors who made this release possible. Here are some highlights ✨:
+
+### \`@mui/x-data-grid@v\` / \`@mui/x-data-grid-pro@v\`
+
+TODO INSERT HIGHLIGHTS
+
+${changeLogMessages.join('\n')}
```

Comment:
The highlights come before the package names.

Reply:
This was not addressed.

![image](https://user-images.githubusercontent.com/42154031/150154982-23ba7be7-b388-4cbe-ae8b-30c0d25bc79a.png)

Reply:
My bad, I moved the "TODO INSERT HIGHLIGHTS" and forget the `changeLogMessages`